dart/lib/src/client.dart

```diff
@@ -188,15 +109,63 @@ abstract class SentryClient {
     options.httpClient?.close();
   }
 
-  @override
-  String toString() => '$SentryClient("$postUri")';
+  SentryEvent _processEvent(
+    SentryEvent event, {
+    dynamic hint,
+    List<EventProcessor> eventProcessors,
+  }) {
+    for (final processor in eventProcessors) {
+      event = processor(event, hint);
+    }
+    return event;
+  }
 
-  @protected
-  List<int> bodyEncoder(Map<String, dynamic> data, Map<String, String> headers);
+  SentryEvent _applyScope(
+      {@required SentryEvent event, @required Scope scope}) {
+    if (scope != null) {
+      // Merge the scope transaction.
+      if (event.transaction == null) {
+        event = event.copyWith(transaction: scope.transaction);
+      }
+
+      // Merge the user context.
+      if (event.userContext == null) {
+        event = event.copyWith(userContext: scope.user);
+      }
+
+      // Merge the scope fingerprint.
+      if (event.fingerprint == null) {
+        event = event.copyWith(fingerprint: scope.fingerprint);
+      }
+
+      // Merge the scope breadcrumbs.
+      if (event.breadcrumbs == null) {
+        event = event.copyWith(breadcrumbs: scope.breadcrumbs);
+      }
+
+      // Merge the scope tags.
+      if (event.tags == null) {
+        event = event.copyWith(tags: scope.tags);
+      }
+
+      // Merge the scope extra.
+      if (event.extra == null) {
+        event = event.copyWith(extra: scope.extra);
+      }
+
+      // Merge the scope level.
+      if (event.level == null) {
+        event = event.copyWith(level: scope.level);
+      }
+    }
```

Comment:
for some fields, we actually append items from the Scope to the event, see:
https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/SentryClient.java#L281-L335

Reply:
✅  fixed